smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs

```diff
@@ -223,28 +257,70 @@ public virtual HttpResponse Request(HttpRequest smartsheetRequest)
                         restRequest.AddHeader(header.Key, header.Value);
                     }
                 }
-
+                Boolean bodyAdded = false;
                 if (smartsheetRequest.Entity != null && smartsheetRequest.Entity.GetContent() != null)
                 {
-                    restRequest.AddParameter(smartsheetRequest.Entity.ContentType, Util.ReadAllBytes(smartsheetRequest.Entity.GetBinaryContent()), 
-                        smartsheetRequest.Entity.ContentType, ParameterType.RequestBody);
+                    //restRequest.AddParameter(smartsheetRequest.Entity.ContentType, Util.ReadAllBytes(smartsheetRequest.Entity.GetBinaryContent()), 
+                    //    smartsheetRequest.Entity.ContentType, ParameterType.RequestBody);
+                    bodyAdded = true;
+                    if (smartsheetRequest.Entity.ContentType == "application/json") {
+                        restRequest = restRequest.AddStringBody(smartsheetRequest.Entity.GetContentAsString(), ContentType.Json);
+                        // JsonParameter param = new JsonParameter();
+                        // restRequest = restRequest.AddBody(param);
+                    } else {
+  
+                        BodyParameter param = new BodyParameter(smartsheetRequest.Entity.ContentType, Util.ReadAllBytes(smartsheetRequest.Entity.GetBinaryContent()), smartsheetRequest.Entity.ContentType);
+                        restRequest = restRequest.AddBody(param);
+                    }
+
                 }
 
                 // Set the client base Url.
-                httpClient.BaseUrl = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
-
+                //httpClient.BaseUrl = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
+                Uri baseUrlAsURI = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
```

Comment:
it looks like these 3ish lines are repeated in several places. it might be helpful to extract a method or function